src/libraries/Microsoft.Bcl.Cryptography/src/System/Security/Cryptography/X509Certificates/X509CertificateKeyAccessors.cs

```diff
@@ -103,14 +112,154 @@ public static class X509CertificateKeyAccessors
         /// <exception cref="InvalidOperationException">
         ///   The certificate already has an associated private key.
         /// </exception>
-        [ExperimentalAttribute(Experimentals.PostQuantumCryptographyDiagId, UrlFormat = Experimentals.SharedUrlFormat)]
+        [Experimental(Experimentals.PostQuantumCryptographyDiagId, UrlFormat = Experimentals.SharedUrlFormat)]
         public static X509Certificate2 CopyWithPrivateKey(this X509Certificate2 certificate, MLKem privateKey) =>
 #if NET10_0_OR_GREATER
             certificate.CopyWithPrivateKey(privateKey);
 #else
             throw new PlatformNotSupportedException(SR.Format(SR.Cryptography_AlgorithmNotSupported, nameof(MLKem)));
 #endif
 
+        /// <summary>
+        ///   Gets the <see cref="MLDsa"/> public key from this certificate.
+        /// </summary>
+        /// <param name="certificate">
+        ///   The X.509 certificate that contains the public key.
+        /// </param>
+        /// <returns>
+        ///   The public key, or <see langword="null"/> if this certificate does not have an ML-DSA public key.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="certificate"/> is <see langword="null"/>.
+        /// </exception>
+        /// <exception cref="PlatformNotSupportedException">
+        ///   The certificate has an ML-DSA public key, but the platform does not support ML-DSA.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   The public key was invalid, or otherwise could not be imported.
+        /// </exception>
+        [Experimental(Experimentals.PostQuantumCryptographyDiagId, UrlFormat = Experimentals.SharedUrlFormat)]
+        public static MLDsa? GetMLDsaPublicKey(this X509Certificate2 certificate)
+        {
+#if NET10_0_OR_GREATER
+            return certificate.GetMLDsaPublicKey();
+#else
+            ArgumentNullException.ThrowIfNull(certificate);
```

Comment:
Good catch, but we should probably do the null check regardless of .NET 10 or not.

Reply:
That would introduce a behavior difference between `certificate.GetMLDsaPublicKey()` and `X509CertificateKeyAccessors.GetMLDsaPublicKey(certificate)` on NET 10. On the other hand, the current code has a behavior difference between NET 9 and NET 10 of `X509CertificateKeyAccessors.GetMLDsaPublicKey(certificate)` (e.g. if people upgrade without recompiling). Not sure what we should prioritize.. what if we just throw `NullReferenceException` so all of them are consistent?

Reply:
FDG's opinion is that the extension method should throw ArgumentNullException.  That is, extension methods should be designed as static methods... they just happen to also support being called via "extension syntax".

So, yeah, someone doing  `cert.GetMLDsaPrivateKey()` will get an ANE on netfx, but an NRE on corefx... but either way, the bug is theirs.

Reply:
Created https://github.com/dotnet/runtime/issues/118066 for follow-up